src/sentry/incidents/logic.py

```diff
@@ -698,17 +687,17 @@ def create_alert_rule(
     return alert_rule
 
 
-def snapshot_alert_rule(alert_rule, user=None):
+def snapshot_alert_rule(alert_rule: AlertRule, user: RpcUser | None = None) -> None:
     # Creates an archived alert_rule using the same properties as the passed rule
     # It will also resolve any incidents attached to this rule.
     with transaction.atomic(router.db_for_write(AlertRuleActivity)):
         triggers = AlertRuleTrigger.objects.filter(alert_rule=alert_rule)
         incidents = Incident.objects.filter(alert_rule=alert_rule)
-        snuba_query_snapshot = deepcopy(alert_rule.snuba_query)
-        snuba_query_snapshot.id = None
+        snuba_query_snapshot: SnubaQuery = deepcopy(_unpack_snuba_query(alert_rule))
+        snuba_query_snapshot.id = None  # type: ignore[assignment]
```

Comment:
This and other `type:ignore`s seem to be unavoidable on various `*.id = None` statements. I don't think I fully grasp the point of nulling out the `id` fields, but it seems to be incompatible with the base model class's declared attributes.

Reply:
hmmm yeah these are a little suspicious -- I believe it's trying to duplicate the row (but get a fresh id) which seems strange -- pretty sure django gives a better way to do this but I don't know it myself

Reply:
if we absolutely have to ignore these errors I would prefer if these were localized somehow instead -- sprinkling type ignores without comments is not great

Reply:
I believe you're correct about the intent. I agree that there's probably a Django utility lurking somewhere to do this and would be glad to go hunting for it later, but prefer to avoid the functional change in this PR.

This calls it out a bit better in the meantime: https://github.com/getsentry/sentry/pull/75541/commits/214ce93d1b4ca95a1769a4661c1f2d0a74af3552

Reply:
See https://github.com/getsentry/sentry/pull/75541/files#r1733575385